atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -300,25 +274,74 @@ def run(self):
                                                             user_provided_release)
 
             if self.reserve_build:
-                self.reserve_build_in_koji(component, version, user_provided_release,
-                                           release_label, dockerfile_labels,
-                                           user_provided_release=True)
+                self.reserve_build_in_koji(
+                    component, version, user_provided_release, explicitly_provided_release=True
+                )
             return
 
-        if release:
-            if not self.append:
-                self.log.debug("release set explicitly so not incrementing")
+        if original_release and not self.append:
+            self.log.debug("release set explicitly so not incrementing")
+            release = original_release
 
-                if not is_scratch_build(self.workflow):
-                    self.check_build_existence_for_explicit_release(component, version, release)
-                    dockerfile_labels[release_label] = release
-                else:
-                    return
-
-        if not release or self.append:
-            self.next_release_general(component, version, release, release_label,
-                                      dockerfile_labels)
+            if not is_scratch_build(self.workflow):
+                self.check_build_existence_for_explicit_release(component, version, release)
+        else:
+            # release not set or release should be appended
+            release = self.next_release_general(component, version, original_release)
```

Comment:
you created commit about ' bump_release: don't pass null release value ', but you are still passing None

Reply:
If `self.append` is True, then original_release may not be None here.

Reply:
yeah I know it is just one of the cases

Reply:
Right, in that commit I meant something more like "don't pass a variable which is guaranteed to be None". I couldn't fit that in the commit title, I hope the rest of the commit message explains it